data/hydraulic-data.py

```diff
@@ -0,0 +1,129 @@
+"""Create hydraulic systems dataset"""
+
+from typing import Union, List
+from h2oaicore.data import CustomData
+import datatable as dt
+import numpy as np
+import pandas as pd
+
+class HydraulicSystemsData(CustomData):
+    # url contains hydraulic systems condition monitoring data set
+    base_url = "https://archive.ics.uci.edu/ml/machine-learning-databases/00447/"
+
+    @staticmethod
+    def create_data(X: dt.Frame = None) -> Union[str, List[str],
+                                                 dt.Frame, List[dt.Frame],
+                                                 np.ndarray, List[np.ndarray],
+                                                 pd.DataFrame, List[pd.DataFrame]]:
+        import os
+        from h2oaicore.systemutils_more import download
+        from h2oaicore.systemutils import config
+        import zipfile
+
+        def mean_conversion(pd_frame):
+            """
+                Means are computed along columns to average the cycle sensor data
+                Each row in frame is data collected over a cycle of 60 seconds
+            """
+            return pd_frame.mean(axis = 1) # axis=1 means along column
+
+        def convert_to_frame(file_bytes_object):
+            """
+                Convert bytes data to pandas dataframe
+            """
```

Comment:
nit/style/optional: There's really no real consensus of how the doc strings should be formatted, but check out the following review comment for a good case for using a slightly different one.

https://github.com/h2oai/model-manager/pull/247#discussion_r413574950

Reply:
Thanks for sharing this helpful link that points to python docs on using docstrings. I believe I made my comments more clear and added keyword arguments to each function.